packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -681,3 +682,22 @@ function getDotDevDotVarsContent(
 
 	return null;
 }
+
+/**
+ * Returns true if the `changedFile` matches one of a potential .dev.vars file.
+ */
+function hasDotDevDotVarsFileChanged(
+	resolvedPluginConfig: ResolvedPluginConfig,
+	changedFile: string
+) {
+	const dotDevDotVarsPaths = new Set(
+		[...resolvedPluginConfig.configPaths].map((configPath) =>
+			path.join(path.dirname(configPath), ".dev.vars")
+		)
+	);
+
+	// Note that we must "resolve" the changed file since the path from Vite will not match Windows backslashes.
```

Comment:
We can resolve the `changedFile` in the hotUpdate hook and the pass the resolved one to  each of the checks.